### DISCUSSION TITLE
Existing j3 issues

### DISCUSSION THREAD

**Discussion by brianteeman** [joomla/joomla-cms#38560] [General] [answered]
Joomla 3 is now in security only mode which, as I understand it, means that there will be no more bug fixes or new features.

There are 120+ open issues flagged as being for J3 which will therefore not be addressed and there is no value in keeping them open.

I propose that
1. They are checked to make sure they do not apply to j4 and if they do they are relabelled
2. If they do not apply to J4 that they are closed with an appropriate message. Perhaps

> Thank you for reporting this issue. Joomla 3 is now in security only mode and as this issue doesn't relate to Joomla 4 it has now been closed. If we are mistaken and this does apply to Joomla 4 please open a new issue (and reference this one if you wish)  with updated details for testing in Joomla 4.

The same text could be used for any new issue created for Joomla 3.

@bembelimen @zero-24 @rdeutz

**Comment by zero-24** [comment, marked as answer]
Yes this is the plan. It had been done for the 3.10-dev PRs but I did not had the time yet to to througth the issues flagged as J3 relevant.

About new PRs we had such message for 2.5 when i remember correctly that was done via the bot in the past and pretty harsh (maybe my memmory fools me on this). Not sure whether that is a good way of doing it? What do you think? Could you propose a text we could use there? Basicly saying the same as for issues but targeted at PRs created against 3.10-dev. As long as its not a massive it could also be done manually i guess.

**Reply by brianteeman** [reply]
1. I would be happy to check and apply the comment to the J3 issues but I dont have the access to close them at the same time
2. Yes I forgot that the same will need to be done for existing PR but there is only one l)
3. A bot could be used for new pull requests against the j3 branch - I would use the same text as I suggested above and just replace "issue" with "pull request"

**Reply by zero-24** [reply]
1. The time intense part is checking whether still relevant or not. When you could start working on that I can do the close work. In that case we might add a info like "will be closed in the comming days"
2. Yes that PR had to be covered too all others had been covered before the last RC release already.
3. Ok will add that bot thing on my todo list. The bot code should be up and running within the issue tracker still it might have to be adjusted for 3.10 including the text.

**Reply by zero-24** [reply]
1. Closing can also be done by the JBS so there are more than one person who can actually close the issues.

**Reply by brianteeman** [reply]
> The time intense part is checking whether still relevant or not. When you could start working on that I can do the close work. In that case we might add a info like "will be closed in the comming days"

ok

**Reply by brianteeman** [reply]
> Closing can also be done by the JBS so there are more than one person who can actually close the issues.

yes they can, I wonder why they dont?

never mind dont answer

**Reply by alikon** [reply]
dunno if just mention @joomla/bug-squad  can speed-up , worth a try

**Reply by alikon** [reply]
@brianteeman every time i've tried to batch close "things" i've always hitted a wall

**Reply by brianteeman** [reply]
you should never batch close things

**Reply by alikon** [reply]
not sure if it is better to have 800 open issues for which no one take care for ages... instead of having few of them, that can reasonably,  worked out from the current work-force

**Reply by zero-24** [reply]
We had that discussion in the past and the agreement was that we are not just closing all issues based on whatever label it has but check whether the issue is still relevant for J4 or not and only close it when the issue is J3 only.

Because when the issues are still relevant they dont "go away" by closing it from github it also does not lower the workload it just hides stuff away and is a bad message to the person who originally reported the issue. The worst case would be someone else has this issue, looks on github, issue closed without any reason or comment -> very bad. Its not better when that issue is still open for years but noone works on that but in that case said person can comment and ask whether it can be picked up again. Noone looks into comments on closed issues and when I'm the person looking for the issue i would not comment too.

But what Brian proposed is different its a look is that an issue still relevant for J4 if yes we keep it open if no we close it with the mentiond message and J3 beeing in sec only mode now.

**Reply by alikon** [reply]
so it's a so far better message to have an issue open for ages  without any feedback,

did you really think that we have the work-power, to go  thru all the open issues,  and check if they are still relevant for j4

i hope you are right

**Reply by zero-24** [reply]
> so it's a so far better message to have an issue open for ages without any feedback,

Yes better than just close all issues without any comment or check.

> did you really think that we have the work-power, to go thru all the open issues, and check if they are still relevant for j4

Yes when we share the work, When I understand brian correct he will start with it. Its also on my todo list also the JBS can work on that. So yes we can work on that step by step. You are right we dont have the work force to do a big bang and check everything today.

> i hope you are right

I hope so too :)

**Reply by brianteeman** [reply]
Yes that is what I proposed. and its really not a big task.

Of course if people kept on top of things and didnt leave things open for years when clearly nothing will be done it would take even less time

**Reply by brianteeman** [reply]
Well in just a day I managed to reduce the list by 50% (and created a few pr in the process)

Its really not hard - you just need to stop looking at page 1 on github

**Reply by zero-24** [reply]
Yes thanks Brian 🎉 :)

**Reply by zero-24** [reply]
The PR for the issue tracker has been done here: https://github.com/joomla/jissues/pull/1172#issue-1351178294

**Comment by alikon** [comment]
thank you 
if they were shares in the market
me as an insider trader .... 😄 
got rich 💵 
![image](https://user-images.githubusercontent.com/181681/186723167-cdde7411-c920-42ee-9c38-68daf5eff536.png)

**Reply by brianteeman** [reply]
Just think how much bettwer it would be if there were more people doing this

**Comment by brianteeman** [comment]
Just 24 left - I think someone else can complete this task ;)

**Reply by zero-24** [reply]
Great thanks Brian :-)
